[bug] regent-gate confirm-feeding-rights returns false after recent dev merges (test regression on main)

## Symptom

Two tests fail in `server/tests/api-downtime-regent-gate.test.js` on the full server suite:

```
AssertionError: expected false to be true
  expect(res.body.feeding_rights_confirmed).toBe(true);
```

Failing tests:
- 'Gate becomes true when all territories with regents have confirmed' (line 156-157)
- 'Territory with no regent_id does not block gate computation' (line 194-195)

**Reproducible:** yes, every run, ~2.0s. Not a flake.

## Where

`POST /api/downtime_cycles/:id/confirm-feeding` — the regent confirms-feeding-rights flow. Both failures: API response's `feeding_rights_confirmed` field is `false` when test expects `true`.

## Pre-existing on main

Verified by Ptah during PR #240 hotfix work. Stashed his changes, re-ran on clean origin/dev — same 2 failures. NOT introduced by #240. Already landed on origin/main before #240's dev → main promote.

## Likely culprit window

Recent merges via Morningstar's lane:

- **#231** 'feat(admin): manual open downtimes override on DT Prep tab' (commit a29e5355) — touches downtime cycle state, plausibly affects gate computation
- **#233** 'fix(dt): wire status/MCI actions into Story tab Status section' (commit 8a71f509) — touches Status/MCI merit actions, may intersect with feeding rights' merit-confirmation flow
- **#236** 'fix(nav): Archive tile always visible in player more-grid' (commit e385917f) — least likely (nav scope)

## Investigation

Either:
1. `git bisect` across #231 / #233 / #236 commits to identify the regression introducer.
2. Ask Morningstar to scope-check those three PRs against the regent-gate test expectations.

## Impact

STs trying to confirm feeding rights for a cycle may see the gate computation report `false` even when all confirmations are in. Regent-gate gating affects feeding-rights distribution.

**User-visible smoke check recommended before next downtime cycle close.**

## Severity

**Medium-to-High** — production-deployed (already on main via #231/#233/#236 merges), API-behaviour bug. Not UI-blocking but adjudication-blocking for regent feeding-rights flow.

## Cross-references

- PR #240 — Mandragora hotfix; surfaced this regression during regression-suite verification, did NOT introduce it.
- Affected file: `server/tests/api-downtime-regent-gate.test.js`

--- a/server/tests/api-downtime-regent-gate.test.js
+++ b/server/tests/api-downtime-regent-gate.test.js
@@ -25,9 +25,36 @@ const OTHER    = new ObjectId().toHexString();
 let insertedCycleIds = [];
 let insertedTerrIds = [];
 
+// Issue #241: tm_suite_test was populated with 5 production-like territories
+// carrying `regent_id` values by the regent-id migration script run on
+// 2026-05-05 (server/scripts/migrate-regent-to-id.js + backup at
+// scripts/_backups/territory-fk-migration-2026-05-05T05-36-59-765Z.json).
+// The confirm-feeding gate at routes/downtime.js:139 queries every
+// territory with `regent_id` set; pre-existing territories leak into the
+// query alongside the test's own inserts and force allConfirmed=false
+// because the test only confirms its single territory. Snapshot the
+// pre-existing regent_id values in beforeAll, null them out for the
+// suite's lifetime, restore in afterAll. The test DB is throwaway so
+// the temporary mutation is bounded to this suite's run; restoring
+// keeps the snapshot intact for subsequent test files in the same
+// vitest worker.
+let _seededRegentSnapshot = [];
+
 beforeAll(async () => {
   await setupDb();
   app = createTestApp();
+  const terrCol = getCollection('territories');
+  _seededRegentSnapshot = await terrCol
+    .find({ regent_id: { $exists: true, $ne: null } })
+    .project({ _id: 1, regent_id: 1, lieutenant_id: 1 })
+    .toArray();
+  if (_seededRegentSnapshot.length) {
+    const ids = _seededRegentSnapshot.map(d => d._id);
+    await terrCol.updateMany(
+      { _id: { $in: ids } },
+      { $set: { regent_id: null, lieutenant_id: null } }
+    );
+  }
 });
 
 afterEach(async () => {
@@ -40,6 +67,19 @@ afterEach(async () => {
 });
 
 afterAll(async () => {
+  // Restore any seeded territories' regent_id we nulled in beforeAll
+  // (issue #241) so subsequent test files in the same worker — and
+  // manual ST smoke runs against tm_suite_test — see the original
+  // state.
+  if (_seededRegentSnapshot.length) {
+    const terrCol = getCollection('territories');
+    for (const doc of _seededRegentSnapshot) {
+      await terrCol.updateOne(
+        { _id: doc._id },
+        { $set: { regent_id: doc.regent_id, lieutenant_id: doc.lieutenant_id ?? null } }
+      );
+    }
+  }
   await teardownDb();
 });
 

